File: src/main/java/com/ject/studytrip/auth/infra/dto/KakaoUserInfoResponse.java
package com.ject.studytrip.auth.infra.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

public record KakaoUserInfoResponse(
        @Schema(description = "카카오 ID") @JsonProperty("id") String kakaoId,
        @Schema(description = "카카오 계정") @JsonProperty("kakao_account") KakaoAccount account) {
    public String getProfileImage() {
        return account.profile().profileImage();
    }

    public String getEmail() {
        return account.email();
    }
}

File: src/main/kotlin/com/ject/studytrip/auth/infra/dto/KakaoUserInfoResponse.kt
package com.ject.studytrip.auth.infra.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class KakaoUserInfoResponse(
    @field:Schema(description = "카카오 ID")
    @field:JsonProperty("id")
    val kakaoId: String,
    @field:Schema(description = "카카오 계정")
    @field:JsonProperty("kakao_account")
    val account: KakaoAccount,
) {
    val profileImage: String
        get() = account.profile.profileImage

    val email: String
        get() = account.email
}
